spotlight-client/src/SystemNarrativePage/SectionNavigation.tsx

```diff
@@ -0,0 +1,171 @@
+  flex-direction: column;
+  height: calc(100vh - ${rem(NAV_BAR_HEIGHT)});
+  justify-content: center;
+  left: ${rem(32)};
+  position: fixed;
+  top: ${rem(NAV_BAR_HEIGHT)};
+`;
+
+const PageNumber = styled.div`
+  font-size: ${rem(13)};
+  font-weight: 600;
+  letter-spacing: -0.01em;
+  line-height: ${rem(16)};
+`;
+
+const PageNumberFaded = styled(PageNumber)`
+  opacity: 0.3;
+`;
+
+const PageProgressContainer = styled.div.attrs(() => ({
+  "aria-hidden": true,
+}))`
+  margin: ${rem(24)} 0;
+  position: relative;
+`;
+
+const PageProgressTrack = styled.div`
+  background: ${colors.rule};
+  height: ${rem(PROGRESS_BAR_HEIGHT)};
+  width: ${rem(2)};
+`;
+
+const PageProgressThumb = styled(animated.div)`
+  background: ${colors.accent};
+  left: 0;
+  position: absolute;
+  width: ${rem(2)};
+`;
+
+const PageProgressBar: React.FC<{
+  currentPage: number;
+  totalPages: number;
+}> = ({ currentPage, totalPages }) => {
+  const thumbSize = PROGRESS_BAR_HEIGHT / totalPages;
+  // pages are 1-indexed for human readability
+  const currentIndex = currentPage - 1;
+
+  const [thumbOffset, setThumbOffset] = useSpring(() => ({
+    top: currentIndex * thumbSize,
+  }));
+
+  useEffect(() => {
+    setThumbOffset({ top: currentIndex * thumbSize });
+  }, [currentIndex, setThumbOffset, thumbSize]);
+
+  return (
+    <PageProgressContainer>
+      <PageProgressTrack />
+      <PageProgressThumb style={{ height: rem(thumbSize), ...thumbOffset }} />
+    </PageProgressContainer>
+  );
+};
+
+type NavigationProps = {
+  activeSection: number;
+  setActiveSection: (section: number) => void;
+  totalPages: number;
+};
+
+const SectionNavigation: React.FC<NavigationProps> = ({
+  activeSection,
+  setActiveSection,
+  totalPages,
+}) => {
+  const { tenantId, narrativeTypeId } = normalizeRouteParams(
+    useParams()
+    // these keys should always be present on this page
+  ) as Required<
+    Pick<
+      ReturnType<typeof normalizeRouteParams>,
+      "tenantId" | "narrativeTypeId"
+    >
+  >;
+
+  const disablePrev = activeSection === 1;
+  const disableNext = activeSection === totalPages;
+
+  // base is the current page, minus the section number
+  const urlBase = getUrlForResource({
+    page: "narrative",
+    params: { tenantId, narrativeTypeId },
+  });
+
+  return (
+    <SectionNav aria-label="page sections">
+      <PageNumber>{formatPageNum(activeSection)}</PageNumber>
+      <PageNumberFaded>{formatPageNum(totalPages)}</PageNumberFaded>
+      <PageProgressBar currentPage={activeSection} totalPages={totalPages} />
+      {disablePrev ? (
+        <div>
+          <Chevron direction="up" faded />
+        </div>
+      ) : (
+        <Link
+          // the to props on these links is mainly for accessibility purposes, so they
+          // look like normal links. calling the setter is the critical step
+          to={`${urlBase}/${activeSection - 1}`}
+          onClick={() => {
+            setActiveSection(activeSection - 1);
```

Comment:
Do you need to add `e.preventDefault()` here too?

Reply:
no actually I should have removed it in both places! and the preventdefault was what was causing the URL history to mysteriously not update in reference to your second comment, which I spent like ....... an hour-plus trying to debug. wow